Buildings/Experimental/DHC/EnergyTransferStations/Combined/Generation5/Controls/Borefield.mo

```diff
@@ -0,0 +1,202 @@
+within Buildings.Experimental.DHC.EnergyTransferStations.Combined.Generation5.Controls;
+model Borefield
+  "Borefield controller"
+  extends Modelica.Blocks.Icons.Block;
+  parameter Modelica.SIunits.Temperature TBorWatEntMax(
+    displayUnit="degC")
+    "Maximum value of borefield water entering temperature";
+  parameter Real spePumBorMin(
+    final unit="1")=0.1
+    "Borefield pump minimum speed";
+  Buildings.Controls.OBC.CDL.Continuous.PIDWithReset conMix(
+    final yMin=0,
+    final yMax=1,
+    final reverseActing=true,
+    y_reset=0,
+    k=0.1,
+    final controllerType=Modelica.Blocks.Types.SimpleController.PI,
+    Ti(
+      displayUnit="s")=120)
+    "Mixing valve controller"
+    annotation (Placement(transformation(extent={{-10,-90},{10,-70}})));
+  Buildings.Controls.OBC.CDL.Continuous.Sources.Constant maxTBorWatEnt(
+    y(final unit="K",
+      displayUnit="degC"),
+    final k=TBorWatEntMax)
+    "Maximum value of borefield water entering temperature"
+    annotation (Placement(transformation(extent={{-50,-90},{-30,-70}})));
+  Buildings.Controls.OBC.CDL.Continuous.GreaterThreshold opeVal(
+    final t=0.9,
+    final h=0.1)
+    "True if at least one isolation valve is open"
+    annotation (Placement(transformation(extent={{-50,-58},{-30,-38}})));
+  Buildings.Controls.OBC.CDL.Continuous.MultiMax multiMax1(
+    nin=2)
+    "Maximum opening"
+    annotation (Placement(transformation(extent={{-90,-50},{-70,-30}})));
+  Buildings.Controls.OBC.CDL.Continuous.GreaterThreshold enaSup(
+    final t=0.05,
+    final h=0.025)
+    "Borefield enabled from supervisory"
+    annotation (Placement(transformation(extent={{-50,-30},{-30,-10}})));
+  Buildings.Controls.OBC.CDL.Logical.Switch runBor
+    "Enable borefield system pump"
+    annotation (Placement(transformation(extent={{70,50},{90,70}})));
+  Buildings.Controls.OBC.CDL.Continuous.Sources.Constant limVal(
+    final k=0.3)
+    "Control signal value for full opening of the valve"
+    annotation (Placement(transformation(extent={{-40,10},{-20,30}})));
+  Buildings.Controls.OBC.CDL.Continuous.Sources.Constant speMin(
+    final k=spePumBorMin)
+    "Minimum pump speed"
+    annotation (Placement(transformation(extent={{-40,74},{-20,94}})));
+  Buildings.Controls.OBC.CDL.Logical.And enaBor
+    "Borefield enable signal"
+    annotation (Placement(transformation(extent={{-10,-40},{10,-20}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput yValIso_actual[2]
```

Comment:
Ideally, move the input/output connector in the section just after parameter declaration. It would be convenient in the debugging process, which can find the interested connector easier in the variable list.

Reply:
Addressed.